ibm-messaging-mq-cloud-showcase-app/backend/AWS-showcase-backend/models/Responder.js

```diff
@@ -34,12 +34,30 @@ class Responder {
             this.mqclient.get(queueName,1,null, 'DYNAMIC', this.appId)
             .then((messages) => {
                 debug_warn(`The producer retrieved this message ${JSON.stringify(messages)}`);
-                if(!messages[0].replyToMsg) {
+                if (messages.length === 0) {
+                    debug_info("Poison Message");
+                    resolve(null);
+                } else if (!messages[0].replyToMsg) {
                     debug_warn("This is not a reply to queue valid message");
                     resolve(null);
+                } else if (messages[0].replyToMsg) {
```

Comment:
This is an array of messages, does the check need to be applied to each message in the array?

Reply:
So I tried out a scenario where a Poison Message is in the queue, and a new fresh Request is sent into the queue. The responder only consumes one message from the queue each time the component is switched "on". So every time a message is consumed, its an array with a single element inside. So that's why we only check for `messages[0]`